keras/layers/core/einsum_dense.py

```diff
@@ -1,5 +1,7 @@
 import re
 
+import numpy as np
```

Comment:
Since einsums can be super general, I added some validation when I added this to KerasNLP.

https://github.com/mattdangerw/keras-nlp/blob/b8d86fe8fe0854b6fa98b7cfb84b73aafa5cd036/keras_nlp/layers/modeling/lora_dense.py#L22-L39

https://github.com/mattdangerw/keras-nlp/blob/b8d86fe8fe0854b6fa98b7cfb84b73aafa5cd036/keras_nlp/layers/modeling/lora_dense.py#L22-L39

We could consider doing something like this? Otherwise I am sure there could equation lines someone constructs that will not work with `enable_lora`.

Reply:
(these are links to the layer we have sense removed for this, but could still mimic the validation)

Reply:
It will always work (unless the kernel is a scalar, but that's impossible), but a caveat is that we only factorize on the last axis, and for weird equations this may not be the desired behavior (it is clearly documented though). For all use cases that I know of, `axis=-1` would be the right behavior.

If you make the axis configurable, things get one order of magnitude more complex, so I didn't expose that option.